### ISSUE TITLE
 integrate an image uploader 

### ISSUE THREAD

**Issue by wynn5a** [BearToCode/carta-md#15]
It is really a great job!  💯 
I'm interested in learning how to integrate an image uploader into this project. 
Would you be able to provide some guidance or ideas on how to proceed?

**Comment by BearToCode** [comment]
Hello!

Here's a basic example of how you could implement that yourself:

```svelte
<script lang="ts">
  // ...

  let dragging = false;
  let uploading = false;

  const imgDropExt: CartaExtension = {
    listeners: [
      ['drop', async function (e) {
        e.preventDefault();

        const file = e.dataTransfer?.files[0];
        if (!file) return;

        uploading = true;

        // Check file type
        // ...
        // Upload the file
        const url = await uploadFile(file);

        // Insert the image
        this.value = this.value.slice(0, this.selectionStart) + `![${file.name}](${url})` + this.value.slice(this.selectionEnd);

        uploading = false;
      }] satisfies CartaListener<'drop'>,
      ['dragenter', () => dragging = true] satisfies CartaListener<'dragenter'>,
      ['dragleave', () => dragging = false] satisfies CartaListener<'dragleave'>,
    ] 
  }

  const carta = new Carta({
    extensions: [imgDropExt],
  });
</script>

<!-- Show some animations for dragging over and uploading -->
```

Anyway, I'm probably going to create a plugin that can handle all of that. Do you have any reference that you would like the end result to look like?

**Comment by wynn5a** [comment]
I'm delighted to know that you'll be implementing the plugin. I believe the GitHub comment editor serves as a valuable reference.
Anticipating the upcoming release of the uploading plugin, especially as I'm a newcomer in the frontend development world. 😃

**Comment by BearToCode** [comment]
Here it is! [plugin-attachment](https://github.com/BearToCode/carta-md/tree/master/packages/plugin-attachment) is now available in version 2.7.0. 

I've not extensively tested the plugin, so feel free to open issues for any problem you may find!
